File: 8kyu/(8 kyu) DNA to RNA Conversion/(8 kyu) DNA to RNA Conversion.cpp
#include <string>
#include <regex>

// #1
// std::string DNAtoRNA(std::string dna)
// {
//     std::regex e("T");
//     std::string result;
//     std::regex_replace(std::back_inserter(result),
//                        dna.begin(), dna.end(), e, "U");
//     return result;
// }

// #2
std::string DNAtoRNA(std::string dna)
{
    std::replace(dna.begin(), dna.end(), 'T', 'U');
    return dna;
}

File: 8kyu/(8 kyu) DNA to RNA Conversion/(8 kyu) DNA to RNA Conversion.rs
fn dna_to_rna(dna: &str) -> String {
    // #1
    // dna.chars()
    //     .map(|x| match x {
    //         'T' => 'U',
    //         _ => x,
    //     })
    //     .collect()
    // #2
    str::replace(dna, "T", "U")
}
